internal/utils/config/config.go

```diff
@@ -43,7 +38,32 @@ func FillDataPlaneProxyContainerEnvs(existing []corev1.EnvVar, podTemplateSpec *
 			container.Env = append(container.Env, envVar)
 		}
 	}
-	for k, v := range envSet {
+	for _, envVar := range envSet {
+		if !k8sutils.IsEnvVarPresent(envVar, container.Env) {
+			container.Env = append(container.Env, envVar)
+		}
+	}
+	sort.Sort(k8sutils.SortableEnvVars(container.Env))
+}
+
+// FillContainerEnvMap updates the environment variables by taking a map of env vars as an input.
+func FillContainerEnvMap(existing []corev1.EnvVar, podTemplateSpec *corev1.PodTemplateSpec, containerName string, envMap map[string]string) {
```

Comment:
This looks identical to `FillContainerEnvs` above, how about reusing `FillContainerEnvs`'s implementation and only converting from the map to the slice in here and calling `FillContainerEnvs`. WDYT?

Reply:
sounds good. I've removed the duplicated function and introduced a conversion helper. PTAL.